src/codecs/namedtuple.ts

```diff
@@ -49,31 +49,24 @@ export class NamedTupleCodec extends Codec implements ICodec, IArgsCodec {
       );
     }
 
-    const keys = Object.keys(args);
-    const names = this.namesSet;
+    const names = this.names;
     const codecs = this.subCodecs;
     const codecsLen = codecs.length;
 
-    if (keys.length !== codecsLen) {
```

Comment:
Good catch! Although I'd keep this check -- we don't want users to supply arguments that aren't used in the query.

Reply:
Would it be useful to say what the extra unused properties are in the error, or just that there are too many?

Reply:
I'd say it would be useful, yes. Please add a test for that.